### PR TITLE
Drop compilers and OpenMPI from conda env generator

### PR DESCRIPTION
**Pull request by manopapad**
@sandeepd-nv @mag1cp1n If we don't pull the compilers from conda anymore, I'm going to need some help on how to adjust the CI scripts (if you think it's necessary).

### CODE REVIEW

**Comment by manopapad** [comment]
I adjusted the approach according to @m3vaz's feedback. Apparently the `--compilers` and `--openmpi` options are still used in CI, so we want to keep them, but we can turn them off by default.

**Review by bryevdv — APPROVED** [review]

File: scripts/generate-conda-envs.py [hunk 1]
```diff
@@ -382,42 +364,39 @@ def __call__(self, parser, namespace, values, option_string):
...
         choices=OS_NAMES,
-        default=None,
-        help="OS to generate for (default: all OSes)",
+        default=("osx" if sys.platform == "darwin" else "linux"),
+        help="OS to generate for",
     )
     parser.add_argument(
         "--compilers",
         action=BooleanFlag,
-        dest="compilers",
```
Comment (m3vaz) [inline scripts/generate-conda-envs.py:404]: Can we retain the `dest` arg for the parse?

File: scripts/generate-conda-envs.py [hunk 2]
```diff
@@ -382,42 +364,39 @@ def __call__(self, parser, namespace, values, option_string):
...
-        default=None,
-        help="Whether to include conda compilers or not (default: both)",
+        default=False,
+        help="Whether to include conda compilers or not",
     )
     parser.add_argument(
         "--openmpi",
         action=BooleanFlag,
-        dest="openmpi",
```
Comment (m3vaz) [inline scripts/generate-conda-envs.py:411]: Can we retain the `dest` arg for the parse?

File: scripts/generate-conda-envs.py [hunk 3]
```diff
@@ -382,42 +364,39 @@ def __call__(self, parser, namespace, values, option_string):
...
-        default=None,
-        help="Whether to include openmpi or not (default: both)",
+        default=False,
+        help="Whether to include openmpi or not",
     )
     parser.add_argument(
         "--ucx",
         action=BooleanFlag,
-        dest="ucx",
```
Comment (m3vaz) [inline scripts/generate-conda-envs.py:418]: Can we retain the `dest` arg for the parse?

**Review by m3vaz — APPROVED** [review]
marking as +1 since the removal of the CTK is just doing some housekeeping
would like to keep the explicit `dest` as well but non-blocking issue

File: scripts/generate-conda-envs.py [hunk 4]
```diff
@@ -276,11 +276,6 @@ def filename(self) -> str:
 
 CTK_VERSIONS = (
     "none",
-    "10.2",
```
[the remark this answers is not in the document]
Comment (m3vaz) [reply scripts/generate-conda-envs.py:279]: this is housekeeping, marking as resolved.

**Pull request merged**